compiler/optimizations/loopInvariantCodeMotion.cpp

```diff
@@ -1068,6 +1085,13 @@ static bool defDominatesAllUses(Loop* loop, SymExpr* def, std::vector<BitVec*>&
  *
  */
 static bool defDominatesAllExits(Loop* loop, SymExpr* def, std::vector<BitVec*>& dominators, std::map<SymExpr*, int>& localMap) {
+  if (def->symbol()->defPoint != nullptr &&
+      LoopStmt::findEnclosingLoop(def->symbol()->defPoint) == loop->getLoopAST()) {
+    // If the symbol-to-hoist is defined inside a loop, no reason to worry about
+    // loop exits, since it can't be referenced outside.
+    return true;
+  }
+
```

Comment:
Is this the part that you're planning to limit to `CHPL_LOCALE_MODEL=gpu` or `isGpuBound`?

Reply:
Yep!